code/game/objects/structures.dm

```diff
@@ -1,25 +1,80 @@
+GLOBAL_LIST_INIT(default_blend_objects,   list(/obj/machinery/door, /turf/simulated/wall))
+GLOBAL_LIST_INIT(default_noblend_objects, list(/obj/machinery/door/window))
+
 /obj/structure
 	icon = 'icons/obj/structures.dmi'
 	w_class = ITEM_SIZE_NO_CONTAINER
 	layer = STRUCTURE_LAYER
 
+	var/handle_generic_blending
+	var/dismantled = FALSE
 	var/breakable
-	var/parts
-	var/list/connections = list("0", "0", "0", "0")
-	var/list/other_connections = list("0", "0", "0", "0")
-	var/list/blend_objects = newlist() // Objects which to blend with
-	var/list/noblend_objects = newlist() //Objects to avoid blending with (such as children of listed blend objects.
-	var/material/material = null
+	var/parts_type
 	var/footstep_type
-	var/mob_offset = 0 //used for on_structure_offset mob animation
+	var/mob_offset
+
+	var/material/material
+	var/material/reinf_material
+	var/material_alteration
+
+/obj/structure/Initialize(var/ml, var/_mat, var/_reinf_mat)
+	if(ispath(_mat, /material))
+		material = _mat
+	if(ispath(material, /material))
+		material = SSmaterials.get_material_datum(material)
+	if(ispath(_reinf_mat, /material))
+		reinf_material = _reinf_mat
+	if(ispath(reinf_material, /material))
+		reinf_material = SSmaterials.get_material_datum(reinf_material)
+	. = ..()
+	if(material || reinf_material)
+		update_materials()
```

Comment:
It's possible this is unsafe until `LateInitialize`; things used to try and update connections from here, which would then check the neighbors' materials, which would not have run the code just above, etc.